plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TrinoCatalog.java

```diff
@@ -117,4 +119,8 @@ void createMaterializedView(
     void renameMaterializedView(ConnectorSession session, SchemaTableName source, SchemaTableName target);
 
     void updateColumnComment(ConnectorSession session, SchemaTableName schemaTableName, ColumnIdentity columnIdentity, Optional<String> comment);
+
+    Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session, SchemaTableName tableName);
+
+    List<SchemaTableName> listTables(ConnectorSession session, SchemaTablePrefix prefix);
```

Comment:
Put this up next to the other `listTables`. Why do we need both?

Reply:
The method is not actually really needed. I have updated the PR to not make use anymore of this method. Affected method `io.trino.plugin.iceberg.IcebergMetadata#streamTableColumns`.